src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php

```diff
@@ -245,12 +246,15 @@ public function indexNodeTranslation(NodeTranslation $nodeTranslation, $add = fa
             return false;
         }
 
+        // Retrieve the referenced entity from the public NodeVersion
+        $page = $publicNodeVersion->getRef($this->em);
+
         $node = $nodeTranslation->getNode();
         if ($this->isIndexable($page)) {
```

Comment:
@treeleaf Maybe use the ReflectionClass here to ? And only when it's indexable you retrieve the ref ?